CorsixTH/Lua/app.lua

```diff
@@ -1215,10 +1215,36 @@ function App:checkInstallFolder()
       " a valid copy of the data files from the original game," ..
       " as said files are required for graphics and sounds."
   if not status then
-    -- If the given directory didn't exist, then likely the config file hasn't
-    -- been changed at all from the default, so we continue to initialise the
-    -- app, and give the user a dialog asking for the correct directory.
-    return false
+    -- Table of predictable places. First three are platform independent,
+    -- then macOS GOG bundle, then linux Filesystem Hierarchy Standard, then Windows
+    local possible_locations = { os.getenv( "HOME" ), os.getenv( "HOME" ) .. pathsep ..  "Documents",
+      select(1, corsixth.require("config_finder")):match("(.*[/\\])"):sub(1, -2),
+      "/Applications/Theme Hospital.app/Contents/Resources/game/Theme Hospital.app/" ..
+        "Contents/Resources/Theme Hospital.boxer/C.harddisk",
+      "/usr/share/games/corsix-th", "/usr/local/share/games/corsix-th",
+      os.getenv("%ProgramFiles%"), os.getenv("%ProgramFiles(x86)%") }
+    local possible_folders = { "ThemeHospital", "Theme Hospital", "HOSP", "TH97",
+      [[GOG.com\Theme Hospital]], [[Origin Games\Theme Hospital\data\Game]] }
+    for _, dir in ipairs(possible_locations) do
+      if status then break end
+      for _, folder in ipairs(possible_folders) do
+        local path = dir .. pathsep .. folder
+        if lfs.attributes(path .. pathsep .. "ANIMS" .. pathsep .. "WINLEVEL.SMK", "size") then
```

Comment:
ANIMS is optional / not part of the CD install. We check for a different file in the directory dialog. May be best to share the implementation.